zebedee/client.go

```diff
@@ -0,0 +1,302 @@
+	if err != nil {
+		return nil, err
+	}
+
+	if len(c.accessToken) > 0 {
+		req.Header.Set("X-Florence-Token", c.accessToken)
+	}
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, ErrInvalidZebedeeResponse{errors.New("unexpected response from zebedee"), zebedeeRequestErrorData{
+			"zebedeeURI":         req.URL.Path,
+			"expectedStatusCode": 200,
+			"actualStatusCode":   resp.StatusCode,
+		}}
+	}
+
+	return ioutil.ReadAll(resp.Body)
+}
+
+func (c ZebedeeClient) getBreadcrumb(uri string) ([]model.TaxonomyNode, error) {
+	b, err := c.get("/parents?uri=" + uri)
+	if err != nil {
+		return nil, err
+	}
+
+	var parentsJSON []models.TaxonomyNode
+	if err = json.Unmarshal(b, &parentsJSON); err != nil {
+		return nil, err
+	}
+
+	var parents []model.TaxonomyNode
+	for _, value := range parentsJSON {
+		parents = append(parents, model.TaxonomyNode{
+			Title: value.Description.Title,
+			URI:   value.URI,
+		})
+	}
+
+	return parents, nil
+}
+
+func (c ZebedeeClient) populateStaticLandingPageModel(dlp models.DatasetLandingPage) (StaticDatasetLandingPage, error) {
+	//Map Zebedee response data to new page model
+	var sdlp StaticDatasetLandingPage
+	sdlp.Type = dlp.Type
+	sdlp.URI = dlp.URI
+	sdlp.Metadata.Title = dlp.Description.Title
+	sdlp.Metadata.Description = dlp.Description.Summary
+	sdlp.DatasetLandingPage.Related.Datasets = dlp.RelatedDatasets
+	sdlp.DatasetLandingPage.Related.Publications = dlp.RelatedDocuments
+	sdlp.DatasetLandingPage.Related.Methodology = append(dlp.RelatedMethodology, dlp.RelatedMethodologyArticle...)
+	sdlp.DatasetLandingPage.Related.Links = dlp.RelatedLinks
+	sdlp.DatasetLandingPage.IsNationalStatistic = dlp.Description.NationalStatistic
+	sdlp.DatasetLandingPage.IsTimeseries = dlp.Timeseries
+	sdlp.ContactDetails.Email = dlp.Description.Contact.Email
+	sdlp.ContactDetails.Telephone = dlp.Description.Contact.Telephone
+	sdlp.ContactDetails.Name = dlp.Description.Contact.Name
+	sdlp.DatasetLandingPage.ReleaseDate = dlp.Description.ReleaseDate
+	sdlp.DatasetLandingPage.NextRelease = dlp.Description.NextRelease
+	sdlp.DatasetLandingPage.Notes = dlp.Section.Markdown
+	sdlp.FilterID = dlp.FilterID
+	bc, err := c.getBreadcrumb(dlp.URI)
+	if err != nil {
+		return StaticDatasetLandingPage{}, err
+	}
+	sdlp.Page.Breadcrumb = bc
+	sdlp.DatasetLandingPage.ParentPath = sdlp.Breadcrumb[len(sdlp.Breadcrumb)-1].Title
+
+	for index, value := range dlp.Datasets {
+		dataset := c.getDatasetDetails(value.URI)
+		dataset.IsLast = index+1 == len(dlp.Datasets)
+
+		sdlp.DatasetLandingPage.Datasets = append(sdlp.DatasetLandingPage.Datasets, dataset)
+	}
+
+	for _, value := range dlp.Alerts {
+		switch value.Type {
+		default:
+			log.Debug("Unrecognised alert type", log.Data{"alert": value})
+			fallthrough
+		case "alert":
+			sdlp.DatasetLandingPage.Notices = append(sdlp.DatasetLandingPage.Notices, datasetLandingPageStatic.Message{
```

Comment:
same here (and a few lines below), might be able to avoid manually mapping the fields, if Date/Markdown are of the same types and there aren't additional fields which could be accidentally copied